### ISSUE TITLE
Since 0.6.0-alpha.1 analyzing does not run in determined order

### ISSUE THREAD

**Issue by web-padawan** [Polymer/polymer-build#97]
### Description
After implementing "analyzer first" approach, unable to use  `polymer-build` as a step in gulp pipeline.
Looks like analysis ignores gulp tasks order at all.

### Versions & Environment
- polymer-build: 0.6.0-alpha.1, 0.6.0-alpha.3
- node: 6.9.1
- Operating System: Mac OS X El Capitan, Windows 7

#### Steps to Reproduce

See [example repo](https://github.com/web-padawan/polymer-sandbox): 

```
git clone https://github.com/web-padawan/polymer-sandbox
npm i
bower i
gulp build
```

For working example, please check [polymer-build-0.5.1](https://github.com/web-padawan/polymer-sandbox/tree/polymer-build-0.5.1) branch.

#### Expected Results
Bundling starts after beginning corresponfing step and ends.

#### Actual Results
Bundling starts before `vulcanize` step beginning and does not finish.
Same can be reproduced with `gulp 4` as well.

@FredKSchott am I doing something wrong here?

**Comment by FredKSchott** [comment]
Thanks for reporting. It looks like polymer-build is just pre-loading your source files (that's the output you're seeing with verbose logging on). Nothing is being analyzed or or bundled until the bundling step is started, as expected.

What kind of problems is this causing for you?

**Comment by web-padawan** [comment]
Actually bundling looks completed with `0.6.0-alpha.3`, but in fact shell does not include some dependencies, e. g. `polymer.html`.

Please compare: 0.6.0-alpha.3 output
```
[21:56:23] Finished 'vulcanize' after 1.82 s
[21:56:23] Starting 'minify:html'...
[21:56:23] minify:html all files 3.2 kB
[21:56:23] Finished 'minify:html' after 32 ms
[21:56:23] Starting 'crisper'...
[21:56:23] crisper all files 3.35 kB
[21:56:23] Finished 'crisper' after 21 ms
[21:56:23] Starting 'minify:js'...
[21:56:23] minify:js all files 395 B
[21:56:23] Finished 'minify:js' after 33 ms
[21:56:23] Starting 'finalize'...
[21:56:23] finalize all files 3.01 kB
[21:56:23] Finished 'finalize' after 13 ms
[21:56:23] Finished 'build' after 3.03 s
```

0.5.1 output
```
[22:01:09] Finished 'vulcanize' after 3.83 s
[22:01:09] Starting 'minify:html'...
[22:01:09] minify:html all files 408.9 kB
[22:01:09] Finished 'minify:html' after 100 ms
[22:01:09] Starting 'crisper'...
[22:01:10] crisper all files 408.46 kB
[22:01:10] Finished 'crisper' after 119 ms
[22:01:10] Starting 'minify:js'...
[22:01:11] minify:js all files 206.95 kB
[22:01:11] Finished 'minify:js' after 1.26 s
[22:01:11] Starting 'finalize'...
[22:01:11] finalize all files 241.08 kB
[22:01:11] Finished 'finalize' after 13 ms
[22:01:11] Finished 'build' after 6.21 s
```

**Comment by FredKSchott** [comment]
I'm sorry I can't reproduce, here's my output when I run with v0.6.0-alpha.3:

```
$ npm ls polymer-build
sandbox@0.0.1 /Users/fschott/Code/polymer-sandbox
└── polymer-build@0.6.0-alpha.3 

$ gulp build
[08:28:53] Using gulpfile ~/Code/polymer-sandbox/gulpfile.js
[08:28:53] Starting 'clean'...
[08:28:53] Finished 'clean' after 169 ms
[08:28:53] Starting 'build'...
[08:28:53] Starting 'index'...
[08:28:53] Starting 'copy'...
[08:28:53] Starting 'fonts'...
[08:28:53] Starting 'images'...
[08:28:53] copy all files 40.55 kB
[08:28:53] Finished 'copy' after 186 ms
[08:28:53] index all files 938 B
[08:28:53] Finished 'index' after 288 ms
[08:28:53] fonts all files 3.15 MB
[08:28:53] Finished 'fonts' after 224 ms
[08:28:54] gulp-imagemin: Minified 2 images (saved 1.97 kB - 5.2%)
[08:28:54] images all files 35.69 kB
[08:28:54] Finished 'images' after 425 ms
[08:28:54] Starting 'vulcanize'...
[08:28:57] Finished 'vulcanize' after 3.32 s
[08:28:57] Starting 'minify:html'...
[08:28:57] minify:html all files 409.64 kB
[08:28:57] Finished 'minify:html' after 112 ms
[08:28:57] Starting 'crisper'...
[08:28:57] crisper all files 409.2 kB
[08:28:57] Finished 'crisper' after 81 ms
[08:28:57] Starting 'minify:js'...
[08:28:58] minify:js all files 207.5 kB
[08:28:58] Finished 'minify:js' after 1.36 s
[08:28:58] Starting 'finalize'...
[08:28:58] finalize all files 241.63 kB
[08:28:58] Finished 'finalize' after 11 ms
[08:28:58] Finished 'build' after 5.42 s
```

Serving the `dist/` build via polyserve works fine as well.

Try doing a full reinstall of your dependencies (`rm -rf node_modules/ && npm i`), does that help?

**Comment by web-padawan** [comment]
Looks fixed on Mac OS, but still can be reproduced on Windows 7 after full reinstall and with using `yarn`

**Comment by FredKSchott** [comment]
@web-padawan It hasn't been fixed yet, Fix is still in review: #99

**Comment by FredKSchott** [comment]
Fixed and merged in #99, we'll publish a new patch release soon
